tun/src/unix/linux/mod.rs

```diff
@@ -129,7 +118,8 @@ impl TunInterface {
         let (connection, handle, _) = new_connection().expect("a new netlink connection");
         let netlink_connection_handle = tokio::task::spawn(connection);
 
-        self.set_iface_up().expect("an active interface");
+        let index = self.index().expect("a valid interface index");
+        _ = handle.link().set(index as u32).up().execute().await;
```

Comment:
I think you were on the right track to separate this out into a different function -- we probably want to control the state of the link independently of the routing tables. What if we're modifying routing before we want the interface up?